makejdk-any-platform.sh

```diff
@@ -40,33 +41,40 @@ for i in "$@"; do
       let counter++
       string="\$$counter"
       OPENJDK_VERSION=$(echo "$@" | awk "{print $string}")
+      if [[ $OPENJDK_VERSION == *u ]]; then
+        OPENJDK_VERSION=${OPENJDK_VERSION::-1}
+        export OPENJDK_VERSION_U="u"
+      fi
+      ;;
+    "--variant" | "-bv")
+      let counter++
+      string="\$$counter"
+      BUILD_VARIANT=$(echo "$@" | awk "{print $string}")
       ;;
   esac
 done
 
 if [ "$OPENJDK_VERSION" == "jdk9" ]; then
   export JDK_PATH="jdk"
   export CONFIGURE_ARGS_FOR_ANY_PLATFORM=${CONFIGURE_ARGS_FOR_ANY_PLATFORM:-"--disable-warnings-as-errors"}
-elif [ "$OPENJDK_VERSION" == "jdk8u" ]; then
+elif [ "$OPENJDK_VERSION" == "jdk8" ]; then
   export JDK_PATH="j2sdk-image"
 else
-  echo "Please specify a version with --version or -v , either jdk9 or jdk8u"
+  echo "Please specify a version with --version or -v , either jdk9 or jdk8, with or without a \'u\' suffix."
```

Comment:
Since OpenJDK [jdk8](http://hg.openjdk.java.net/jdk8) and [jdk8u](http://hg.openjdk.java.net/jdk8u/jdk8u/) are different code repos, I find it confusing to make `--version jdk8u` a synonym for `--version jdk8`.  What's the rationale for not requiring the trailing 'u'?

Reply:
jdk9 is soon to become 9u, and extensions for jdk 8 and 9 don't have a "u" in their repository name.

I thought this was simpler, solved the future 9u problem, and also allows the user to decide if they want a u while not totally relinquishing control.

Reply:
Plus, if someone just copies the build script arguments for jdk8u and sticks "-variant=openj9" in there, it removes this problem for them.

Reply:
@tellison raises a reasonable point. Since this "not having a u" thing is specific to the OpenJ9 Extensions repositories, maybe the variables could be constructed differently to reflect that.

Perhaps OPENJDK_VERSION should remain the actual full "forest" name (jdk8, jdk8u, jdk9, jdk9u, etc.) and we introduce an OPENJDK_VERSION_BASELEVEL or something like that, to denote the corresponding jdk8, jdk8, jdk9, jdk9 (respectively), and we use that variable to set the OpenJ9 extensions repository?

That's just a suggestion; @tellison @gdams @karianna @neomatrix369 any opinions?

Reply:
That makes more sense to me yes. Making an implicit derivation of an environment variable into its own explicit declaration matches the convention of the code in this repo.

@adamfarley Thanks again for your patience on this, appreciate it's frustrating have good functionality held up in review!.

Reply:
Thanks karianna.

If we don't seperate the "u" like this, I worry we'll be in for another problem when jdk9u (mercurial) happens, not to mention jdk10->jdk10u, jdk11->jdk11u, etc.

Reply:
We could separate that into another issue I guess. Trying to solve too many problems at once, perhaps.

Reply:
@adamfarley yep, let's think ahead, but not try to fix the future just yet :-)  Just make the current logic as clear as we can so it's easy to change for the next person.

So as @mstoodle wrote, defining extra variables with clear names that denote what they are will be better than manipulating a smaller set with append/pre-pend knowledge.
Though IMHO
> OPENJDK_VERSION should remain the actual full "forest" name" 

it may be better to rename the variable `OPENJDK_FOREST_NAME` etc if we go that route to maintain maximum readability.

Reply:
Ok, done and comitted.

Reply:
I'm going to quibble a bit on the choice of OPENJDK_FOREST_NAME, just because in JDK10 the "forest" has actually been collapsed (that's why I put it in quotes originally :) ). But I don't care enough to block it because I can still squint and accept it; I'm just registering I don't think that name is 100% accurate either, so if we're going to change it, I'd suggest changing it to OPENJDK_REPO_NAME (sorry @adamfarley :) ). I am willing to leave that to the JDK10 support work so this can move forward.

Reply:
@mstoodle Can you add that to #189 ?